ethereum/core/src/main/java/tech/pegasys/teku/core/stategenerator/AsyncChainStateGenerator.java

```diff
@@ -153,9 +169,9 @@ public static AsyncChainStateGenerator create(
                   ChainStateGenerator.create(chainBlocks, startState, true);
               final AtomicReference<BeaconState> lastState = new AtomicReference<>(null);
               chainStateGenerator.generateStates(
-                  (block, state) -> {
-                    lastState.set(state);
-                    handler.handle(block, state);
+                  (stateAndBlock) -> {
```

Comment:
nit: Remove unnecessary parenthesis:
```suggestion
                  stateAndBlock -> {
```